plugins/octopus-deploy/src/components/ScaffolderDropdown/ProjectGroupDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  InputLabel,
+  Input,
+  FormControl,
+  FormHelperText,
+} from '@material-ui/core';
+import { Select, SelectItem } from '@backstage/core-components';
+import { useProjectGroups } from '../../hooks/useProjectGroups';
+import { ProjectGroupDropdownProps } from './schema';
+
+export const ProjectGroupDropdown = ({
+  onChange,
+  rawErrors,
+  required,
+  formData,
+}: ProjectGroupDropdownProps) => {
+  const [selectedProjectGroup, setSelectedProjectGroup] =
+    React.useState<string>('');
+  const { projectGroups } = useProjectGroups();
+
+  const projectGroupItems: SelectItem[] = projectGroups
+    ? projectGroups.map(pg => ({ label: pg.Name, value: pg.Name }))
+    : [];
+
+  const updateProjectGroup = (pg: string) => {
+    setSelectedProjectGroup(pg);
+    onChange(pg);
+  };
+
+  return (
+    <FormControl
```

Comment:
There's a `ScaffolderField` component that you can use here to refactor away some of the boilerplate for a Field Extension.

https://github.com/backstage/backstage/blob/d202a258d181e0ff7e726de30e105107ca9e4647/plugins/scaffolder-react/src/next/components/ScaffolderField/ScaffolderField.tsx#L53

Should make this a little cleaner.

Reply:
Refactored it to use the ScaffolderField now :)